code/game/objects/items/devices/spyglasses.dm

```diff
@@ -1,64 +1,305 @@
+	top_border = temp_list["top"]
+	var/icon/target_icon = getFlatIcon(target)
+	var/attempts = 0
+	var/selected_x
+	var/selected_y
+	var/selected_coordinates = FALSE
+	while(attempts < 10)
+		selected_x = rand(left_border + 2, right_border - 2)
+		selected_y = rand(bottom_border + 2, top_border - 2)
+		if(isnull(target_icon.GetPixel(selected_x, selected_y)))
+			attempts ++
+		else
+			selected_coordinates = TRUE
+			break
+	if(!selected_coordinates)
+		selected_x = 15
+		selected_y = 15
+	user.temporarilyRemoveItemFromInventory(src)
+	forceMove(null)
+	attached_to = target
+	tracker = new /datum/movement_detector(target, CALLBACK(src, PROC_REF(update_view)))
+	target.tracking_bug = src
+	target.tracking_bug.x_coord = selected_x - 15
+	target.tracking_bug.y_coord = selected_y - 15
+	target.update_appearance(updates = UPDATE_ICON|UPDATE_OVERLAYS)
+	update_view()
+	return TRUE
+
+/obj/item/video_bug/proc/detach_from_clothing(mob/user, var/obj/item/clothing/target)
+	target.tracking_bug.attached_to = null
 	tracker = new /datum/movement_detector(src, CALLBACK(src, PROC_REF(update_view)))
+	target.tracking_bug.update_view()
+	target.tracking_bug = null
+	target.update_appearance(updates = UPDATE_ICON|UPDATE_OVERLAYS)
 
+/obj/item/video_bug/attackby(obj/item/I, mob/living/user, params)
+	. = ..()
+	if(istype(I, /obj/item/clothing/glasses/sunglasses/spy))
+		var/obj/item/clothing/glasses/sunglasses/spy/spyglasses
+		if(spyglasses.bug_network_id)
+			bug_network_id = spyglasses.bug_network_id
+			to_chat(user,"<span class='notice'>You connect \the [src] to \the [spyglasses]'s network!</span>")
+		else if(bug_network_id)
+			spyglasses.bug_network_id = bug_network_id
+			to_chat(user,"<span class='notice'>You connect \the [spyglasses] to \the [src]'s network!</span>")
+
+/obj/item/video_bug/attack_self(mob/user)
+	. = ..()
+	if(being_viewed)
+		to_chat(user, "<span class='warning'>\The [src] is currently in use and its ID cannot be changed!</span>")
+		return
+	var/picked_id = tgui_input_text(user, "Please enter an ID for this tracker.", "ID ", bug_id)
+	if(picked_id)
+		bug_id = picked_id
+		to_chat(user, "<span class='notice'>You set \the [src]'s ID to [picked_id].</span>")
+
+/obj/item/video_bug/pre_attack(atom/A, mob/living/user, params)
+	var/succeeded = FALSE
+	if(istype(A, /obj/item/clothing))
+		if(attach_to_clothing_item(user, A, TRUE))
+			to_chat(user, "<span class='notice'>You stealthily plant \the [src] on \the [A]!</span>")
+			succeeded = TRUE
+		else
+			return ..()
+	if(iscarbon(A) && !succeeded)
+		if(attach_to_carbon(user, A))
+			succeeded = TRUE
+	if(!succeeded)
+		return ..()
+
+/obj/item/video_bug/emp_act(severity)
+	if(severity > 4)
+		Destroy(src)
+		return ..()
+	disabled = TRUE
+	update_appearance(updates = UPDATE_ICON|UPDATE_OVERLAYS)
+	if(attached_to)
+		attached_to.update_appearance(updates = UPDATE_ICON|UPDATE_OVERLAYS)
+	update_view()
+	addtimer(CALLBACK(src, PROC_REF(recover_from_emp)), 60 SECONDS * severity)
```

Comment:
You need to verify this but I think that 1 severity is high severity and 2 is low severity.